src/com/jetbrains/SharedTextures.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2000-2025 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.jetbrains;
+
+import java.awt.*;
+
+/**
+ * The service provides functionality for working with shared textures in JetBrainsRuntime.
+ */
+@Service
+@Provided
+public interface SharedTextures {
+    /**
+     * Shared textures are not supported.
+     */
+    public final int NotSupported = 0;
+    /**
+     * Metal textures are supported.
+     */
+    public final int MetalTextureType = 1;
+
+    /**
+     * Checks what type of texture is supported by the current rendering pipeline.
+     *
+     * @return the type of shared texture supported, or {@link SharedTextures#NotSupported}
+     * if shared textures are not supported.
+     */
+    int getTextureType();
```

Comment:
Can there be multiple values, e.g. bitmask?

Reply:
Also, is `NotSupported` really needed? Looks like we can just report the whole service as unsupported in this case.

Reply:
That actually a good question. Don't know, to be honest. 
In the foreseeable future just one of Metal, D3D(SharedHandle), Vulkan, OpenGL, probably IOSurface…

Reply:
This is certainly possible to have two or more backends here (think Vulkan and memory buffers for Wayland), but it will never be possible to have two at the same time. So in this sense, returning a singular value is fine.

Reply:
Since it(multiple texture types) is not supported and not planned, I left it as it is.
If we need it some day, the API could be extended.